test/utils.py

```diff
@@ -79,3 +80,47 @@ def wait_file(dir_name, file_pattern, exclude_list, timeout_sec=1):
         iter_count = iter_count + 1
 
     return ""
+
+
+def create_test_instance(instance_cfg_file_name, instance_cfg_file_dir, tmpdir):
+    """Create test tarantool instance via subprocess.Popen with given cfg file.
+
+    Performs this steps for it:
+    1) Copy the instance config files to the run pytest tmp directory.
+       This cfg file should be in /test/integration/foo-module/test_file/instance_cfg_file_name.
+       But you can specify different path by instance_cfg_file_dir arg.
+
+    2) Also copy to pytest tmpdir the lua module utils.lua with the auxiliary functions.
+       This functions is required for using with your instance.
+       As a result, you can use require('utils') inside your instance config file.
+
+    3) Run tarantool via subprocess.Popen with given cfg file.
+       Gets bound port from tmpdir.
+       Return subprocess object and instance's port.
+
+    NOTE: Demand require('utils').bind_free_port(arg[0]) inside your instance cfg file.
+
+    Args:
+        instance_cfg_file_name (str): file name of your test instance cfg.
+        instance_cfg_file_dir (str): path to dir that contains instance_cfg_file_name.
+        tmpdir (str): expected result of fixture get_tmpdir from conftest.
+
+    Returns:
+        Popen[bytes]: subprocess.Popen object with tarantool test instance.
+        str: port of tarantool test instance.
+    """
+
+    # Copy the instance config files to the run pytest tmpdir directory.
+    shutil.copy(instance_cfg_file_dir + "/" + instance_cfg_file_name, tmpdir)
+    # Copy the lua module with the auxiliary functions required by the instance config file.
+    shutil.copy(instance_cfg_file_dir + "/../../../" + "utils.lua", tmpdir)
+    # Start test instance for testing.
+    instance_obj = subprocess.Popen(["tarantool", instance_cfg_file_name], cwd=tmpdir)
+    # The delay is needed so that the instance has time to start and configure itself.
+    time.sleep(1)
```

Comment:
Yeah, not a nice thing.

It is better to spin in a loop around some condition: say, presence of the file with the address to connect. Or in a loop with attempt to read the file and connect to the address. I also suggest to spin here until a timeout: otherwise hard to debug 'hangs' may appear.

I can guide you around the similar code in test-run, which run, I guess, around 100k tests per day (so any race in it is pretty visible). You can glance over the code around https://github.com/tarantool/test-run/pull/348 and https://github.com/tarantool/test-run/pull/349 or ask me to present it.

Reply:
Done, implemented 3 sec timeout with existence of port file check